### ISSUE TITLE
Custom language server not starting

### ISSUE THREAD

**Issue by sayingandparsing** [neoclide/coc.nvim#656]
vim version: NVIM v0.4.0-dev
node version: v10.15.3
coc.nvim version: 0.0.64
term: xterm-256color
platform: linux

I'm trying to setup a custom language service running on node. I installed the `vim-node-rpc` module globally using yarn, and created on entry for in `coc-settings.json` (see below), but it doesn't start when editing the specified file types.
In the log, I see:
```
[services] - registed service "languageserver.natural-language-server"
[plugin] - coc initialized with node: v10.15.3
```
but I never get a message like `service XYZ started`, as with other services.
Additionally, there're no logged error, and I know my server code is never executed since it's logs are empty.
I'm clearly missing something from the configuration, but haven't been able to figure out what it is.

Config file:
```json
{
  "languageserver": {
    "natural-language-server": {                                                                                                                                                                                                              
      "module": "/home/reagan/code/proj/natural-language-server/out/server.js",                                                                                                                                                               
      "args": ["--node-ipc"],                                                                                                                                                                                                                 
      "filetypes": ["txt", "md"],                                                                                                                                                                                                             
      "trace.server": "verbose",                                                                                                                                                                                                              
      //"rootPatterns": ["root.yml"],                                                                                                                                                                                                         
      "execArgv": ["--nolazy", "--inspect-brk=6045"],                                                                                                                                                                                         
      "initializationOptions": {                                                                                                                                                                                                              
      },                                                                                                                                                                                                                                      
      "settings": {                                                                                                                                                                                                                           
        "validate": true                                                                                                                                                                                                                      
      }                                                                                                                                                                                                                                       
    }                                                                                                                                                                                                                                         
  }                                                                                                                                                                                                                                           
}  
```

**Comment by chemzqm** [comment]
` "execArgv": ["--nolazy", "--inspect-brk=6045"]` this would break your node process on startup.

**Comment by chemzqm** [comment]
Remove that unless you want to debug your languageserver.

**Comment by sayingandparsing** [comment]
@chemzqm I removed that line from the configuration but am still having the same issue. Is there anything else I should try?

**Comment by chemzqm** [comment]
Can you make sure your filetype is txt or md?  Use `:CocList services` to get service status.

**Comment by sayingandparsing** [comment]
Yeah, it duplicates the list items for some reason, but it's registered from those. I get:
`languageserver.natural-language-server [init] txt, txt, md, md`

Is there any way I can start it manually?

**Comment by chemzqm** [comment]
There is action of list and api to start LS service.

**Comment by sayingandparsing** [comment]
Thanks, it starts the module when I select it from CocList. The only problem now is that it's not sending completion requests. My modules responds to the initialization request, at which point coc logs a summary of the capabilities.
```
        capabilities: {
            textDocumentSync: 1,
            completionProvider: {
                resolveProvider: false
            },
            documentOnTypeFormattingProvider: {
                firstTriggerCharacter: ' '
            }
        }
```
Coc then sends a DidChangeConfiguration notification.
However, after that, no further requests are issued in response to typing. 

I created both a completion provider and an onTypeFormatting provider. Do I need to somehow specify the conditions under which completions are requested, or does the Coc client default to send them following each typed character.
Not sure if the onTypeFormatting provider will work since I used a space as the trigger character, but for now I'm mostly worried about completion
